code/SimpleLight.dm

```diff
@@ -92,3 +91,155 @@ atom
+					medium_light.color = rgb(rgba[1], rgba[2], rgba[3], min(255, rgba[4]))
+				else
+					// divided by two because the directional sprites are brighter
+					medium_light.color = rgb(rgba[1], rgba[2], rgba[3], min(255, rgba[4] / 2))
+		else
+			update_medium_light_color()
+
+
+	proc/remove_medium_light(var/id)
+		if (!medium_light_rgbas)
+			return
+
+		if (id)
+			if (id in medium_light_rgbas)
+				//medium_light_rgbas -= medium_light_rgbas[id]
+				medium_light_rgbas.Remove(id)
+		else
+			medium_light_rgbas.len = 0
+
+		if (medium_light_rgbas.len <= 0)
+			hide_medium_light()
+		else
+			update_medium_light_color()
+
+	proc/update_medium_light_color()
+		var/avg_r = 0
+		var/avg_g = 0
+		var/avg_b = 0
+		var/sum_a = 0
+
+		for (var/id in medium_light_rgbas)
+			avg_r += medium_light_rgbas[id][1]
+			avg_g += medium_light_rgbas[id][2]
+			avg_b += medium_light_rgbas[id][3]
+			sum_a += medium_light_rgbas[id][4]
+
+		avg_r /= medium_light_rgbas.len
+		avg_g /= medium_light_rgbas.len
+		avg_b /= medium_light_rgbas.len
+
+		for(var/obj/overlay/simple_light/medium/medium_light in src.medium_lights)
+			if(medium_light.icon_state == "medium_center")
+				medium_light.color = rgb(avg_r, avg_g, avg_b, min(255, sum_a))
+			else
+				// divided by two because the directional sprites are brighter
+				medium_light.color = rgb(avg_r, avg_g, avg_b, min(255, sum_a / 2))
+
+	proc/show_medium_light()
+		if (!medium_lights)
+			medium_lights = list()
+			for(var/light_dir in src.medium_light_dirs)
+				var/obj/overlay/simple_light/medium/light = new(null, light_dir)
+				src:vis_contents += light
+				src.medium_lights += light
+		for(var/obj/overlay/simple_light/medium/light in src.medium_lights)
+			light.invisibility = 0
+		update_medium_light_visibility()
+
+	proc/hide_medium_light()
+		for(var/obj/overlay/simple_light/medium/light in src.medium_lights)
+			light.invisibility = 101
+
+	proc/destroy_medium_light()
+		if (medium_light_rgbas && medium_light_rgbas.len)
+			hide_medium_light()
+		for(var/obj/overlay/simple_light/medium/light in src.medium_lights)
+			src:vis_contents -= light
+			qdel(light)
+		medium_light_rgbas = null
+		src.medium_lights = null
+
+	disposing()
+		..()
+		if (src.medium_lights)
+			destroy_medium_light()
+
+	proc/update_medium_light_visibility()
+		if(src.medium_lights[1].invisibility == 101) // toggled off
+			return
+		if(!istype(src.loc, /turf))
+			for(var/obj/overlay/simple_light/medium/light in src.medium_lights)
+				light.invisibility = 102
+			return
+		for(var/obj/overlay/simple_light/medium/light in src.medium_lights)
+			if(light.icon_state == "medium_center")
+				light.invisibility = 0
+				continue
+			var/turf/T = get_step(get_turf(src), light.dir)
+			if(T.opacity || T.opaque_atom_count)
+				light.invisibility = 102
+			else
+				light.invisibility = 0
+
+/atom/proc/add_sm_light(id, list/rgba, medium=null)
+	// TODO decide cleverly whether medium or simple if medium == null
```

Comment:
TODO

Reply:
Alright, replaced with a placeholder that probably doesn't work correctly.